lib/pages/settings/view/admin_page.dart

```diff
@@ -0,0 +1,102 @@
+import 'package:acs_upb_mobile/generated/l10n.dart';
+import 'package:acs_upb_mobile/pages/settings/service/admin_provider.dart';
+import 'package:acs_upb_mobile/pages/settings/view/request_card.dart';
+import 'package:acs_upb_mobile/widgets/scaffold.dart';
+import 'package:acs_upb_mobile/widgets/toast.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_feather_icons/flutter_feather_icons.dart';
+import 'package:provider/provider.dart';
+
+class AdminPanelPage extends StatefulWidget {
+  const AdminPanelPage({Key key}) : super(key: key);
+  static const String routeName = '/admin';
+
+  @override
+  _AdminPanelPageState createState() => _AdminPanelPageState();
+}
+
+class _AdminPanelPageState extends State<AdminPanelPage> {
+  Future<List<String>> requests;
+  List<String> requestIds;
+  bool all = false;
+
+  @override
+  void initState() {
+    super.initState();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final adminProvider = Provider.of<AdminProvider>(context);
```

Comment:
I think this is the reason the page rebuilds entirely. Try to do listen: false here.

(the only time the whole page needs to rebuild is when the filter is changed, and you're doing that by calling `setState` already; one could also argue that a request should "disappear" once approved if you have "show unprocessed" selected, but I think it's better to show it with the revert button instead, at least until the page is opened later or the filter is changed)